include/galsim/Silicon.h

```diff
@@ -249,6 +256,12 @@ namespace galsim
 
         void updatePixelBounds(int nx, int ny, size_t k);
 
+        void updatePixelBoundsGPU(int nx, int ny, size_t k,
+                                  Bounds<double>* pixelInnerBoundsData,
+                                  Bounds<double>* pixelOuterBoundsData,
+                                  Position<float>* horizontalBoundaryPointsData,
+                                  Position<float>* verticalBoundaryPointsData);
```

Comment:
Let's not have two ways to do this please.  Rewrite the existing updatePixelBounds to work with the GPU.  Don't repeat everything in a second function.